packages/compose/src/hooks/use-disabled/index.js

```diff
@@ -50,56 +54,54 @@ const DISABLED_ELIGIBLE_NODE_NAMES = [
  * ```
  */
 export default function useDisabled() {
-	/** @type {import('react').RefObject<HTMLElement>} */
-	const node = useRef( null );
+	return useRefEffect( ( node ) => {
+		const disable = () => {
+			node.style.setProperty( 'user-select', 'none' );
+			node.style.setProperty( '-webkit-user-select', 'none' );
+			focus.focusable.find( node ).forEach( ( focusable ) => {
+				if (
+					includes( DISABLED_ELIGIBLE_NODE_NAMES, focusable.nodeName )
+				) {
+					focusable.setAttribute( 'disabled', '' );
+				}
 
-	const disable = () => {
-		if ( ! node.current ) {
-			return;
-		}
+				if ( focusable.nodeName === 'A' ) {
+					focusable.setAttribute( 'tabindex', '-1' );
+				}
 
-		focus.focusable.find( node.current ).forEach( ( focusable ) => {
-			if (
-				includes( DISABLED_ELIGIBLE_NODE_NAMES, focusable.nodeName )
-			) {
-				focusable.setAttribute( 'disabled', '' );
-			}
+				const tabIndex = focusable.getAttribute( 'tabindex' );
+				if ( tabIndex !== null && tabIndex !== '-1' ) {
+					focusable.removeAttribute( 'tabindex' );
+				}
 
-			if ( focusable.nodeName === 'A' ) {
-				focusable.setAttribute( 'tabindex', '-1' );
-			}
+				if ( focusable.hasAttribute( 'contenteditable' ) ) {
+					focusable.setAttribute( 'contenteditable', 'false' );
+				}
 
-			const tabIndex = focusable.getAttribute( 'tabindex' );
-			if ( tabIndex !== null && tabIndex !== '-1' ) {
-				focusable.removeAttribute( 'tabindex' );
-			}
+				if (
+					node.ownerDocument.defaultView?.HTMLElement &&
+					focusable instanceof
+						node.ownerDocument.defaultView.HTMLElement
+				) {
+					focusable.style.setProperty( 'pointer-events', 'none' );
```

Comment:
This is another addition in this PR. Basically any clickable item inside the disabled element is now "non clickable"